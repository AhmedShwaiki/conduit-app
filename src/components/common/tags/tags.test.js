import toJson from "enzyme-to-json";
import { shallow, mount } from "enzyme";
import { React } from "react";
import Tags from "./tags";
import { BrowserRouter as Router } from "react-router-dom";

describe("<Tags/>", () => {
  it("tags renders correctly", () => {
    const wrapper = mount(
      <Router>
        <Tags items={["tag1"]} />
      </Router>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
