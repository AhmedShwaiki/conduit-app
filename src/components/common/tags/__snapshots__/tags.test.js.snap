// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tags/> tags renders correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Tags
      items={
        Array [
          "tag1",
        ]
      }
    >
      <styled.div>
        <div
          className="sc-bdfBwQ hwgBUR"
        >
          <p>
            Popular Tags
          </p>
          <styled.div>
            <div
              className="sc-gsTCUz yshIc"
            >
              <Styled(Link)
                item="tag1"
                key="tag1"
                onClick={[Function]}
                to="#"
              >
                <Link
                  className="sc-dlfnbm kFiSyS"
                  item="tag1"
                  onClick={[Function]}
                  to="#"
                >
                  <LinkAnchor
                    className="sc-dlfnbm kFiSyS"
                    href="/"
                    item="tag1"
                    navigate={[Function]}
                    onClick={[Function]}
                  >
                    <a
                      className="sc-dlfnbm kFiSyS"
                      href="/"
                      item="tag1"
                      onClick={[Function]}
                    >
                      tag1
                    </a>
                  </LinkAnchor>
                </Link>
              </Styled(Link)>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </Tags>
  </Router>
</BrowserRouter>
`;
